pages/api/workouts/index.js

```diff
@@ -1,27 +1,39 @@
 import Workout from "@/db/models/Workout";
 import dbConnect from "@/db/connect.js";
+import { getToken } from "next-auth/jwt";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../auth/[...nextauth]";
 
 export default async function handler(request, response) {
+  const session = await getServerSession(request, response, authOptions);
   await dbConnect();
 
   if (request.method === "GET") {
-    try {
-      const workouts = await Workout.find();
+    if (session) {
+      const token = await getToken({ req: request });
+      const userId = token.sub;
```

Comment:
you can reduce repetition by using an early return (if no session available) and then once use the getToken method.